spec/services/migration/migrators/declaration_superseded_by_spec.rb

```diff
@@ -0,0 +1,36 @@
+require "rails_helper"
+
+RSpec.describe Migration::Migrators::DeclarationSupersededBy do
+  it_behaves_like "a migrator", :declaration_superseded_by, %i[declaration] do
+    def create_ecf_resource
+      create(:ecf_migration_participant_declaration).tap do |declaration|
+        # We use the same declaration here to make the test setup easier.
+        declaration.update!(superseded_by_id: declaration.id)
+      end
+    end
+
+    def create_npq_resource(ecf_resource)
+      lead_provider = create(:lead_provider, ecf_id: ecf_resource.cpd_lead_provider.npq_lead_provider.id)
+      cohort = create(:cohort, start_year: ecf_resource.cohort.start_year)
+      course = create(:course, identifier: ecf_resource.course_identifier)
+      user = create(:user, ecf_id: ecf_resource.user.id)
+      application = create(:application, :accepted, course:, user:)
+      create(:declaration, ecf_id: ecf_resource.id, cohort:, lead_provider:, application:)
```

Comment:
could we make this ineligible so it matches superseded declarations incase in ECF?

Reply:
not sure I follow you here? there's actually a bunch of setup that's not needed in this migrator I've now removed (as the migrator simply matches on the `ecf_id` and updates the `superseeded_by` it doesn't care about cohort/user/application)

Reply:
ah sorry meant that a superseded declaration is always ineligible, if we show it in the test it helps indicate why those are ineligible thats all :)